mesa/examples/basic/schelling/agents.py

```diff
@@ -16,15 +15,19 @@ def __init__(self, model, agent_type: int) -> None:
 
     def step(self) -> None:
         """Determine if agent is happy and move if necessary."""
-        neighbors = self.model.grid.iter_neighbors(
+        neighbors = self.model.grid.get_neighbors(
             self.pos, moore=True, radius=self.model.radius
         )
 
         # Count similar neighbors
-        similar = sum(neighbor.type == self.type for neighbor in neighbors)
+        similar_neighbors = len([n for n in neighbors if n.type == self.type])
```

Comment:
The model looks significantly slower. I’m curious if we could minimize that performance degradation, I suspect the main problem is with this line doing additional work.

Reply:
The slowdown is not due to this line of code but because the behavior changes a lot. Prior to this PR, the model would reach steady state in a few ticks. Now, it can take many ticks and frequently the model never reaches stable state.